src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,72 @@
 package core.basesyntax;
 
+import static java.lang.Integer.parseInt;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final int OPERATION_INDEX = 0;
+    private static final int VALUE_INDEX = 1;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String dataFromFile = readFromFile(fromFileName);
+        String report = getReport(dataFromFile);
+        writeToFile(report, toFileName);
+    }
+
+    private String readFromFile(String fromFileName) {
+        File file = new File(fromFileName);
+        String value = null;
+        try {
+            BufferedReader bufferedReader = new BufferedReader(new FileReader(file));
+            StringBuilder stringBuilder = new StringBuilder();
+            value = bufferedReader.readLine();
+            while (value != null) {
+                stringBuilder.append(value).append(System.lineSeparator());
+                value = bufferedReader.readLine();
+            }
+            String data = stringBuilder.toString();
+            return data;
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read from a file", e);
+        }
+    }
+
+    private String getReport(String data) {
+        String supply = "supply";
+        int supplyValue = 0;
+        int buyValue = 0;
+        int result = 0;
+        String[] dataFromFile = data.split(System.lineSeparator());
+        for (int i = 0; i < dataFromFile.length; i++) {
+            String[] operation = dataFromFile[i].split(",");
+            int secondPart = parseInt(operation[VALUE_INDEX]);
+            if (operation[OPERATION_INDEX].equals(supply)) {
+                supplyValue += secondPart;
+            } else {
+                buyValue += secondPart;
+            }
+        }
+        result = supplyValue - buyValue;
+        StringBuilder report = new StringBuilder();
+        report.append("supply,").append(supplyValue).append(System.lineSeparator())
+                .append("buy,").append(buyValue).append(System.lineSeparator())
+                .append("result,").append(result);
+        return report.toString();
+    }
 
+    public void writeToFile(String report, String toFileName) {
+        try {
+            BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(toFileName));
+            bufferedWriter.write(report);
+            bufferedWriter.close();
```

Comment:
writer should be closed in block 'finally'. But it would be better to use try with resources here